hazelcast/src/main/java/com/hazelcast/query/impl/InternalIndex.java

```diff
@@ -41,6 +42,13 @@
      */
     boolean hasPartitionIndexed(int partitionId);
 
+    /**
+     * Returns {@code true} if all {@code queryPartitions} are indexed,
+     * {@code false} otherwise.
+     * @param queryPartitions a set of partitions a query runs on.
+     */
+    boolean allPartitionsIndexed(PartitionIdSet queryPartitions);
```

Comment:
given that we only need owned partition count for comparison, this signature seemed clearer:
`boolean allPartitionsIndexed(int ownedPartitionCount)`

Reply:
Also can we add to javaDoc that this method has meaning with only `global indexes`? + a short description of solved race issue would be good in this methods javaDoc.